saw-core/src/Verifier/SAW/Simulator/Prims.hs

```diff
@@ -649,41 +617,47 @@ atWithDefaultOp bp =
           VVector xv -> force (vecIdx d xv (fromIntegral i)) -- FIXME dangerous fromIntegral
           VWord xw -> VBool <$> bpBvAt bp xw (fromIntegral i) -- FIXME dangerous fromIntegral
           _ -> panic "atOp: expected vector"
-      VToNat i -> do
+      VBVToNat _sz i -> do
         iv <- toBits (bpUnpack bp) i
         case x of
           VVector xv ->
-            selectV (lazyMuxValue bp) (fromIntegral n - 1) (force . vecIdx d xv) iv -- FIXME dangerous fromIntegral
+            selectV (lazyMuxValue bp tp) (fromIntegral n - 1) (force . vecIdx d xv) iv -- FIXME dangerous fromIntegral
           VWord xw ->
-            selectV (lazyMuxValue bp) (fromIntegral n - 1) (liftM VBool . bpBvAt bp xw) iv -- FIXME dangerous fromIntegral
+            selectV (lazyMuxValue bp tp) (fromIntegral n - 1) (liftM VBool . bpBvAt bp xw) iv -- FIXME dangerous fromIntegral
           _ -> panic "atOp: expected vector"
+
+      VIntToNat _i ->
+        error "atWithDefault: symbolic integer TODO"
```

Comment:
We'll probably need to implement a proper indexing operation with symbolic integer index.